coderd/database/dbfake/dbfake.go

```diff
@@ -2349,17 +2418,20 @@ func (q *FakeQuerier) GetTemplateDailyInsights(ctx context.Context, arg database
 	q.mutex.RLock()
 	defer q.mutex.RUnlock()
 
-	type dailyStat struct {
+	type statByInterval struct {
 		startTime, endTime time.Time
 		userSet            map[uuid.UUID]struct{}
 		templateIDSet      map[uuid.UUID]struct{}
 	}
-	dailyStats := []dailyStat{{arg.StartTime, arg.StartTime.AddDate(0, 0, 1), make(map[uuid.UUID]struct{}), make(map[uuid.UUID]struct{})}}
-	for dailyStats[len(dailyStats)-1].endTime.Before(arg.EndTime) {
-		dailyStats = append(dailyStats, dailyStat{dailyStats[len(dailyStats)-1].endTime, dailyStats[len(dailyStats)-1].endTime.AddDate(0, 0, 1), make(map[uuid.UUID]struct{}), make(map[uuid.UUID]struct{})})
+
+	interval := time.Duration(arg.IntervalDays) * 24 * time.Hour
+
+	statsByInterval := []statByInterval{{arg.StartTime, arg.StartTime.Add(interval), make(map[uuid.UUID]struct{}), make(map[uuid.UUID]struct{})}}
```

Comment:
I think this could still use `.AddDate`, the reason I point it out is caution with DST. E.g. does it behave the same as an interval when crossing the DST boundary?

I _think_ Go assumes this for you, but I think it would be good to confirm 👍🏻

So the question is: Does the clock stay `00:00` when using `Add(interval)` and DST is part of time range?